### PR TITLE
Swap arguments to Grad and Hess

### PR DESCRIPTION
**Pull request by vladimir-ch**
@btracey PTAL

### CODE REVIEW

File: types.go [hunk 1]
```diff
@@ -127,11 +127,11 @@ type Problem struct {
 
 	// Grad evaluates the gradient at x and stores the result in-place in grad.
 	// Grad must not modify x.
-	Grad func(x []float64, grad []float64)
+	Grad func(grad []float64, x []float64)
 
 	// Hess evaluates the Hessian at x and stores the result in-place in hess.
 	// Hess must not modify x.
-	Hess func(x []float64, hess *mat64.SymDense)
+	Hess func(hess *mat64.SymDense, x []float64)
```
Comment (btracey) [inline types.go]: Can we change this to MutableSym while we're breaking everything?

**Comment by btracey** [comment]
LGTM ignoring the interface change.

File: types.go (hunk 1 above)
Comment (vladimir-ch) [inline types.go]: AFAICT we don't have MutableSym (yet).

Comment (btracey) [inline types.go]: https://github.com/gonum/matrix/pull/242

Comment (vladimir-ch) [inline types.go]: Argh ... and @kortschak complains that _he_ says foolish things ...

**Pull request merged**